alf/optimizers/optimizers.py

```diff
@@ -92,8 +166,99 @@ def step(self, closure=None):
             else:
                 tensor_utils.clip_by_norms(
                     grads, self._gradient_clipping, in_place=True)
+
+        if self._parvi is not None:
+            for param_group in self.param_groups:
+                if "parvi_grad" in param_group:
+                    params = param_group['params']
+                    batch_size = params[0].shape[0]
+                    params_tensor = torch.cat(
+                        [p.view(batch_size, -1) for p in params],
+                        dim=-1)  # [N, D], D=dim(params)
+                    if self._parvi == 'svgd':
```

Comment:
I thought we could not do svgd this way? Need some explanation.

Reply:
Let us discuss this further offline.